internal/bminventory/inventory.go

```diff
@@ -2297,3 +2309,22 @@ func computeClusterProxyHash(httpProxy, httpsProxy, noProxy *string) (string, er
 	bs := h.Sum(nil)
 	return fmt.Sprintf("%x", bs), nil
 }
+
+func validateProxySettings(httpProxy, httpsProxy, noProxy *string) error {
+	if httpProxy != nil && *httpProxy != "" {
+		if err := validations.ValidateHTTPProxyFormat(*httpProxy); err != nil {
+			return common.NewApiError(http.StatusBadRequest, err)
```

Comment:
I think you can jsut return `fmt.Errorf` you wrap it as http.StatusBadRequest in the parent method.
Also consider adding to the error which proxy setting validation failed (http/https)

Reply:
Fixed: 
for https proxy validation error it produces:
"Failed to validate HTTPS Proxy: Proxy URL format is not valid: 'http://wwww:asdsa@10.46.41.7a***.a:3128'"

and for http proxy validation error:
"Failed to validate HTTP Proxy: Proxy URL format is not valid: 'http://**10.46.41.7:3128'"